apps/qubit/modules/informationobject/actions/editAction.class.php

```diff
@@ -451,7 +451,16 @@ protected function processField($field)
               break;
 
             case 'relatedMaterialDescriptions':
-              $relation->typeId = QubitTerm::RELATED_MATERIAL_DESCRIPTIONS_ID;
+              // Don't allow self-relations
+              if (!isset($this->resource->id) || ($this->resource->id != $item->id))
+              {
+                $relation->typeId = QubitTerm::RELATED_MATERIAL_DESCRIPTIONS_ID;
+              }
+              else
+              {
+                $message = $this->context->i18n->__('Self-relation ignored.');
+                $this->getUser()->setFlash('notice', $message);
+              }
```

Comment:
Hi @mcantelon, is the relation not being created or it's being created without a type id after this change? It looks like the type id is not a required column, so I guess it's being created, which is not ideal.

Reply:
Yikes... yeah, that's definitely not optimal. I'll fix.

Reply:
Nice! thanks!